dnsapi/dns_zoneee.sh

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env sh
+
+ZONEEE_api="https://api.zone.eu/v2/dns"
+
+# Usage: add  _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+# Used to add txt record
+dns_zoneee_add() {
+  fulldomain=$1
+  txtvalue=$2
+
+  ZONEEE_User="${ZONEEE_User:-$(_readaccountconf_mutable ZONEEE_User)}"
+  ZONEEE_Key="${ZONEEE_Key:-$(_readaccountconf_mutable ZONEEE_Key)}"
+  if [ -z "$ZONEEE_User" ] || [ -z "$ZONEEE_Key" ]; then
+    ZONEEE_User=""
+    ZONEEE_Key=""
+    _err "You haven't specified zone.ee user and api key."
+    _err "Please Add your credentials and try again."
+    _err "Username as ZONEEE_User."
+    _err "API key as ZONEEE_Key."
+    return 1
+  fi
+
+  #save the credentials to the account conf file.
+  _saveaccountconf_mutable ZONEEE_User "$ZONEEE_User"
+  _saveaccountconf_mutable ZONEEE_Key "$ZONEEE_Key"
+
+  _debug "First detect the root zone"
+  if ! _get_root "$fulldomain"; then
+    _err "invalid domain"
+    return 1
+  fi
+
+  _debug _sub_domain "$_sub_domain"
+  _debug _domain "$_domain"
+
+  _debug "Getting txt records"
+  _zoneee_rest GET "$_domain/txt"
+
+  if printf "%s" "$response" | grep \"result\":error >/dev/null; then
```

Comment:
please use `echo` instead of `printf %s`  here.

Reply:
Done, and improved the error message a bit as well.